sklearn/preprocessing/data.py

```diff
@@ -1956,3 +1967,222 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class CountFeaturizer(BaseEstimator, TransformerMixin):
+
+    """
+    Adds in a new feature column containing the number of occurences of
+    each row in the given data set.
+
+    Formally, for each data point X_i in the dataset X, it will add in
+    a new feature count_X_i to the end of X_i where count_X_i is the number of
+    occurences of X_i in the dataset X given the equality indicator
+    'inclusion'
+
+    This preprocessing step is useful when the number of occurences
+    of a particular piece of data is helpful in computing the prediction.
+
+    Parameters
+    ----------
+    inclusion: set, list, numpy.ndarray, or string (only 'all'), default='all'
```

Comment:
need space before colon

I'd be interested in supporting a list of lists of features, and an easy way for each feature to be counted independently.